decentraland/unity-explorer
# fix: third party wearables & emote urns and fixes on wearable resolving process

## What does this PR change?

Third party wearables and emotes were not working due its urn structure.
TP do not include the tokenId and has 7 parts, so we must not shorten them. Example: `urn:decentraland:matic:collections-thirdparty:dolcegabbana-disco-drip:0x4bd77619a75c8eda181e3587339e7011da75bf0e:2a424e9c-c6fb-4783-99ed-63d260d90ed2`.

Also, some TPW wearable urns comes in UPPER CASE, but `entities/active` endpoint gets everything lowered provoking cache misses in the content. This change makes the urns case-insensitive.
Finally added some safe checks on resolving wearables so the avatars do not fail when a DTO is not resolved.

## How to test the changes?

1. Equip a third party wearable and/or emote into your avatar
2. Check that works as expected
3. Go to the backpack, change some wearables, keeping the TPW
4. Check that the avatar is updated in-world
5. You will not be able to equip TPW since we do not support the filters in the backpack yet. You will have to equip it through the old renderer.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
